src/solver/simulation/solver.hxx

```diff
@@ -281,7 +279,9 @@ inline ISimulation<Impl>::ISimulation(Data::Study& study,
 
     pHydroHotStart = (study.parameters.initialReservoirLevels.iniLevels == Data::irlHotStart);
 
-    study.setWriter(pResultWriter);
+    // Initialize writer
+    study.resultWriter = resultWriterFactory(study.parameters.resultFormat, study.folderOutput, pQueueService, duration_collector);
+    pResultWriter = study.resultWriter;
```

Comment:
When `study` is instantiated, `pQueueService` doesn't exist yet since it is instantiated by `simulation` object of type `ISimulation<Impl>` (e.g **src/solver/main/economy.cpp**)